server/plugin/api.go

```diff
@@ -640,12 +641,16 @@ func (p *Plugin) addCommentsOnRecord(w http.ResponseWriter, r *http.Request) {
 func (p *Plugin) getStatesForRecordType(w http.ResponseWriter, r *http.Request) {
 	pathParams := mux.Vars(r)
 	recordType := pathParams[constants.PathParamRecordType]
-	if !constants.ValidRecordTypesForSearching[recordType] {
-		p.API.LogError("Invalid record type while trying to get the record", "Record type", recordType)
+	if !constants.RecordTypesSupportingStateUpdation[recordType] {
```

Comment:
```suggestion
	if !constants.RecordTypeSupportingStateUpdation[recordType] {
```
should not we remove 's' from the 'types' as at a time it's targeting/working for only 1 type?

Reply:
That variable is a map that contains all the record types that support state updation. I don't think removing the 's' will be a better name for it.